weed/shell/command_volume_fsck.go

```diff
@@ -126,20 +139,25 @@ func (c *commandVolumeFsck) Do(args []string, commandEnv *CommandEnv, writer io.
 		return fmt.Errorf("read filer buckets path: %v", err)
 	}
 
-	collectMtime := time.Now().UnixNano()
+	c.collectStartTime = time.Now()
 	// collect each volume file ids
 	for dataNodeId, volumeIdToVInfo := range dataNodeVolumeIdToVInfo {
 		for volumeId, vinfo := range volumeIdToVInfo {
-			if *c.volumeId > 0 && uint32(*c.volumeId) != volumeId {
-				delete(volumeIdToVInfo, volumeId)
-				continue
+			if len(c.volumeIds) > 0 {
+				if _, ok := c.volumeIds[volumeId]; !ok {
+					delete(volumeIdToVInfo, volumeId)
+					continue
+				}
 			}
-			if *c.collection != "" && vinfo.collection != *c.collection {
+			// or skip /topics/.system/log without collection name
+			if (*c.collection != "" && vinfo.collection != *c.collection) || vinfo.collection == "" {
```

Comment:
this is not in sync with master. This `vinfo.collection == ""` is removed. It would always skip data in default collection.

Reply:
> this is not in sync with master. This `vinfo.collection == ""` is removed. It would always skip data in default collection. 


This is the point(hotfix), since in each pass of fsck in the collection vinfo.collection == ""( /topics/.system/log)  there are orphaned files

Reply:
For example, if a cluster does not use any collections, this command would not work.

Reply:
> For example, if a cluster does not use any collections, this command would not work.

Rechecked with an empty weed server and the problem did not reproduce. removed `vinfo.collection == ""`

Reply:
You need to think about how to separate files without collections from files with collections during fsck https://github.com/seaweedfs/seaweedfs/issues/3912